packages/embed/src/js/metadata/user-agent.js

```diff
@@ -0,0 +1,63 @@
+export const parseUserAgent = (uaString) => {
+  const parser = {
+    browser: '',
+    browserVersion: '',
+    os: '',
+    osVersion: '',
+    device: '',
+  };
+
+  // Browser
+  const browserRegex =
+    /(firefox|msie|trident|chrome|safari|opera|edge|edg|opr|brave|vivaldi)/i;
+  const browserMatch = uaString.match(browserRegex);
+
+  if (browserMatch) {
+    parser.browser = browserMatch[1].toLowerCase();
+
+    // Browser version
+    const versionRegex = new RegExp(
+      `${parser.browser}[\\s/](\\d+(\\.\\d+)*)`,
+      'i',
+    );
+    const versionMatch = uaString.match(versionRegex);
+
+    if (versionMatch) {
+      parser.browserVersion = versionMatch[1];
+    }
+  }
+
+  // Operating System
+  const osRegex = /(windows|macintosh|mac os x|linux|android|ios|iphone|ipad)/i;
+  const osMatch = uaString.match(osRegex);
+
+  if (osMatch) {
+    parser.os = osMatch[1].toLowerCase();
+
+    // OS version
+    const osVersionRegex = new RegExp(
+      `${parser.os}[\\s/](\\d+(\\.\\d+)*)`,
```

Comment:
The os version structure can be different for different os
eg for macos 

"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.2 Safari/605.1.15"

this has under score rather than dot

We might need to add the os specific logic to get the version